src/main/java/org/openrewrite/gradle/RecipeMarketplaceCsvValidateCompletenessTask.java

```diff
@@ -121,4 +113,24 @@ void validate() {
 
         getLogger().lifecycle("Recipe marketplace CSV completeness validation passed");
     }
+
+    /**
+     * Construct an Environment that only loads recipes directly from the given recipe JAR, not from its dependencies.
+     * This ensures that completeness validation is only done against recipes actually provided by the recipe JAR.
+     */
+    private Environment jarScanningEnvironment(Path recipeJarPath) {
+        // Get runtime classpath, as that contains classes needed to load recipes
+        JavaPluginExtension javaExtension = getProject().getExtensions().getByType(JavaPluginExtension.class);
+        List<Path> classpath = javaExtension.getSourceSets()
+                .getByName("main")
+                .getRuntimeClasspath()
+                .getFiles()
+                .stream()
+                .map(File::toPath)
+                .filter(path -> path.toString().endsWith(".jar")) // Exclude build output directories
+                .toList();
+
+        // Load environment from JAR
+        return Environment.builder().scanJar(recipeJarPath, classpath, new RecipeClassLoader(recipeJarPath, classpath)).build();
```

Comment:
Previously we'd load the recipes without these dependencies, which caused us to not be able to load any classes that depended on classes defined elsewhere, like in other language modules or even third party libraries like the SQL formatter we use. This should fix that for all such cases, removing the need to add dependencies to the plugin as we'd earlier done for rewrite-templating.